platform/mellanox/docker-syncd-mlnx/ecmp_calc_sdk.py

```diff
@@ -0,0 +1,466 @@
+            # Calculate sonic index (sonic index=4 for Ethernet4)
+            lane_index = get_lane_index(lane_bmap, ASIC_MAX_LANES[chip_type])
+            assert lane_index != -1, "Failed to calculate port index"
+            
+            sonic_index = label_port * ASIC_MAX_LANES[chip_type] + lane_index;
+            sonic_interface = ETHERNET_PREFIX + str(sonic_index)    
+            ports_map[logical_port] = sonic_interface
+            
+        return ports_map
+    
+    finally:
+        delete_uint32_t_p(port_cnt_p)
+        delete_sx_port_attributes_t_arr(port_attributes_list)
+
+def sx_router_get_ecmp_id(handle, vrid, ip_prefix):
+    """ Get ECMP id for a given IP prefix
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+        vrid (sx_router_id_t): Virtual router id
+        ip_prefix (sx_ip_prefix_t): Network address 
+    
+    Returns:
+        int: ECMP id
+    """
+    try:
+        ip_prefix_p = new_sx_ip_prefix_t_p()
+        sx_ip_prefix_t_p_assign(ip_prefix_p, ip_prefix)
+        entries_cnt_p = new_uint32_t_p()
+        uint32_t_p_assign(entries_cnt_p, 1)
+        entries_array = new_sx_uc_route_get_entry_t_arr(1)
+        
+        rc = sx_api_router_uc_route_get(handle, SX_ACCESS_CMD_GET, vrid, ip_prefix_p, None, entries_array, entries_cnt_p)
+        if rc == SX_STATUS_ENTRY_NOT_FOUND:
+            return 0
+        sx_check_rc(rc)
+        
+        entry = sx_uc_route_get_entry_t_arr_getitem(entries_array, 0)
+        if entry.route_data.type == SX_UC_ROUTE_TYPE_NEXT_HOP:
+            return entry.route_data.uc_route_param.ecmp_id
+        
+        return 0
+    
+    finally:
+        delete_sx_ip_prefix_t_p(ip_prefix_p)
+        delete_uint32_t_p(entries_cnt_p)
+        delete_sx_uc_route_get_entry_t_arr(entries_array)
+
+def sx_router_operational_ecmp_get(handle, ecmp_id):
+    """ Get next hops for a given ECMP id
+    
+    Args:
+        handle (sx_api_handle_t): SDK handle
+        ecmp_id (int): ECMP id
+    
+    Returns:
+        list: List of next hops
+    """    
+    try:
+        next_hops = []
+        
+        next_hop_count_p = new_uint32_t_p()
+        uint32_t_p_assign(next_hop_count_p, 0)
+        
+        rc = sx_api_router_operational_ecmp_get(handle, ecmp_id, None, next_hop_count_p)
+        sx_check_rc(rc)
+        
+        next_hop_count  = uint32_t_p_value(next_hop_count_p)
+        next_hop_list_p = new_sx_next_hop_t_arr(next_hop_count)
+        
+        rc = sx_api_router_operational_ecmp_get(handle, ecmp_id, next_hop_list_p, next_hop_count_p)
+        sx_check_rc(rc)
+        
+        next_hop_count  = uint32_t_p_value(next_hop_count_p)
+        for i in range(next_hop_count):
+            next_hop = sx_next_hop_t_arr_getitem(next_hop_list_p, i)
+            if next_hop.next_hop_key.type == SX_NEXT_HOP_TYPE_IP:
+                next_hops.append(next_hop)
+                
+        return next_hops
+    
+    finally:
+        delete_uint32_t_p(next_hop_count_p)
```

Comment:
@liorghub free is opposite to construction

Reply:
Done.